### PR TITLE
Add tests for and update documentation for tagged snippets

### PR DESCRIPTION
**Pull request by timheap**
Following the [Tagging recipe](http://docs.wagtail.io/en/v1.0/reference/pages/model_recipes.html#tagging) does not work for snippets. Instead, you should use TaggableManager. A test case checking that tags work on Snippets has been added, and the documentation will be updated shortly.

### CODE REVIEW

**Comment by gasman** [comment]
We previously encountered the same problem with tags not saving on pages, and the workaround then was to call form.save() twice: https://github.com/torchbox/wagtail/issues/197#issuecomment-49982808

It sounds like the same workaround needs to be applied to the snippets edit view. But fixing the root cause would be better, of course!

File: wagtail/tests/testapp/models.py [hunk 1]
```diff
@@ -302,14 +302,22 @@ class AdvertPlacement(models.Model):
...
+class AdvertTag(TaggedItemBase):
+    content_object = ParentalKey('Advert', related_name='tagged_items')
+
+
 @python_2_unicode_compatible
 class Advert(models.Model):
     url = models.URLField(null=True, blank=True)
     text = models.CharField(max_length=255)
 
+    tags = ClusterTaggableManager(through=AdvertTag, blank=True)
```
Comment (kaedroho) [inline wagtail/tests/testapp/models.py]: Have you tried with ``TaggableManager``?

**Comment by timheap** [comment]
Heh, the tests I had written were completely screwed up after the rebase, but I didn't have time to test and fix them before I left work. Fixed them now!

Using TaggableManager instead of ClusterTaggableManager works just fine. I'll update the documentation to state when to use which one for which models, and submit it as part of this PR

**Comment by SalahAdDin** [comment]
:+1:

File: docs/topics/snippets.rst [hunk 2]
```diff
@@ -169,3 +169,28 @@ These child objects are now accessible through the page's ``advert_placements``
...
+    from taggit.models import TaggedItemBase
+    from taggit.managers import TaggableManager
+
+    class AdvertTag(TaggedItemBase):
+        content_object = ParentalKey('demo.Advert', related_name='tagged_items')
+
+    @register_snippet
+    class Advert(models.Model):
+        ...
+        tags = ClusterTaggableManager(through=BlogPageTag, blank=True)
```
Comment (kaedroho) [inline docs/topics/snippets.rst]: Typo: ``ClusterTaggableManager`` => ``TaggableManager``

**Comment by timheap** [comment]
Updated the documentation. I have no idea why the travis build is failing. It worked last time, and I didn't change anything except the docs. It is complaining about a conflicting migration, but one of the migrations doesnt actually exist...

File: wagtail/wagtailsnippets/tests.py [hunk 3]
```diff
@@ -90,6 +92,20 @@ def test_create(self):
...
+                                        'url': 'http://example.com/',
+                                        'tags': ', '.join(tags)})
+
+        self.assertRedirects(response, reverse('wagtailsnippets:list',
+                                               args=('tests', 'advert')))
+
+        snippet = Advert.objects.get(text='test_advert')
+        self.assertEqual(
+            list(snippet.tags.order_by('name')),
+            list(Tag.objects.order_by('name').filter(name__in=tags)))
```
Comment (gasman) [inline wagtail/wagtailsnippets/tests.py]: We should also test that these lists are non-empty (since this assertion would also pass if the tags field was ignored completely).

**Comment by timheap** [comment]
Thanks!
